nomenclature/testing.py

```diff
@@ -144,6 +144,7 @@ def assert_valid_structure(
         )
 
     dsd = DataStructureDefinition(path / definitions, dimensions)
+    dsd.variable.data_validator
```

Comment:
I think this will fail if we don't have the variable dimension. Apart from this, I think the line technically works but only in a somewhat hard to read way. I'd make it more explicit and therefor easier to read, something like:
```python
if dsd.variable:
    DataValidator.from_codelist(dsd.variable)
```